Mathlib/Data/Real/Sqrt.lean

```diff
@@ -478,6 +485,19 @@ instance : StarOrderedRing ℝ :=
 
 end Real
 
+instance NNReal.instStarOrderedRing : StarOrderedRing ℝ≥0 where
+  le_iff x y := by
+    constructor; swap
+    · rintro ⟨p, -, rfl⟩
+      exact le_self_add
+    rw [← NNReal.coe_le_coe, StarOrderedRing.le_iff]
+    rintro ⟨p, hp, h⟩
+    refine ⟨⟨p, (AddSubmonoid.closure_le.2
+      (range_subset_iff.2 fun x ↦ ?_) hp : p ∈ AddSubmonoid.nonneg ℝ)⟩, ?_, NNReal.coe_injective h⟩
+    · simp [mul_self_nonneg]
+    · exact AddSubmonoid.subset_closure
+        ⟨NNReal.sqrt _, by dsimp; rw [star_trivial, ← sqrt_mul, sqrt_mul_self]⟩
+
```

Comment:
I'm pretty sure there's a helper constructor that lets you go from a square root function.

Reply:
Cannot find it 😐 @j-loreaux ?

Reply:
`StarOrderedRing.ofNonnegIff'` will help

Reply:
That doesn't apply since `ℝ≥0` isn't a ring. You would need to have something like `NonUnitalCancelSemiring` to be able to state the generalisation that applies here.